apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/sharing-modal/use-add-tags-to-post.ts

```diff
@@ -0,0 +1,28 @@
+import apiFetch from '@wordpress/api-fetch';
+
+type HasAddedTagsResult = {
+	added_tags: number;
+	success: boolean;
+};
+
+// OnSaveTags is a function callback that takes the number of tags added
+// to the post as an argument.
+type OnSaveTagsCallback = ( addedTags: number ) => void;
+const useAddTagsToPost = ( postId: number, tags: string[], onSaveTags: OnSaveTagsCallback ) => {
+	async function saveTags() {
+		try {
+			const result: HasAddedTagsResult = await apiFetch( {
+				method: 'POST',
+				path: `/wpcom/v2/read/sites/${ window._currentSiteId }/posts/${ postId }/tags/add`,
+				data: { tags },
+			} );
+			onSaveTags( result.added_tags ?? 0 );
+		} catch ( error ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Error: Unable to add tags. Reason: %s', error );
```

Comment:
This doesn't format the error which is an object, you could JSON.stringify or log error.message

Reply:
also, it should ideally display an error notification to the user

Reply:
I have tried to handle this case a bit better, I now return number of tags added - if error caught, this will be `0` - we then show notice that 'No Tags Added' which I feel is enough here.